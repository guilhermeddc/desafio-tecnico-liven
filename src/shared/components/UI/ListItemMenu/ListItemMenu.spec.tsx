import {render, screen} from '@testing-library/react';

import {ListItemMenu} from '.';

jest.mock('react-router-dom', () => ({
  useLocation: jest.fn(),
}));

describe('ListItemMenu', () => {
  it('renders correctly', () => {
    const {container} = render(
      <ListItemMenu label="Home" link="/" onClick={() => null} />,
    );

    expect(container).toMatchSnapshot();
  });

  it('renders correctly with props', () => {
    render(<ListItemMenu label="Home" link="/" onClick={() => null} />);

    expect(screen.getByText('Home')).toBeInTheDocument();
  });
});
