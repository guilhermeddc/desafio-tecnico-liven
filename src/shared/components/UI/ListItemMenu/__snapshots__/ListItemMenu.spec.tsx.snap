// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListItemMenu renders correctly 1`] = `
<div>
  <div
    class="MuiButtonBase-root MuiListItem-root MuiListItem-gutters MuiListItem-padding MuiListItem-button css-11r3gg3-MuiButtonBase-root-MuiListItem-root"
    tabindex="0"
    to="/"
  >
    <div
      class="MuiListItemText-root css-tlelie-MuiListItemText-root"
    >
      <h6
        class="MuiTypography-root MuiTypography-h6 MuiTypography-alignCenter css-xjy3u8-MuiTypography-root"
      >
        Home
      </h6>
    </div>
    <span
      class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
    />
  </div>
</div>
`;
